shim/shim.c

```diff
@@ -432,6 +446,7 @@ handle_proxy_output(struct cc_shim *shim)
 		code = *(buf + STREAM_HEADER_SIZE); 	// hyperstart has sent the exit status
 		shim_debug("Exit status for container: %d\n", code);
 		free(buf);
+		restoreTerminal();
```

Comment:
Thanks - yes, since there is already a signal handler, the `atexit`-registered function will be called for those signals already handled.

It's worth noting though that those signals that are currently not explicitly handled and which default to terminate will leave the terminal in raw mode. We should probably document those somewhere since someone is bound to try something like...

```
docker kill --signal SIGSEGV $container
```
... and wonder why their terminal is screwed.

`^Jstty sane^J` to the rescue! :smile: